src/reducers/importText.ts

```diff
@@ -107,14 +107,15 @@ const importText = (
     const textNormalized = strip(convertedText, { preserveFormatting: true })
 
     // insert the textNormalized into the destValue in the correct place
-    // trim after concatenating in case destValue has whitespace
-    const left = (destValue.slice(0, replaceStart ?? 0) + textNormalized).trimLeft()
     // if cursorCleared is true i.e. clearThought is enabled we don't have to use existing thought to be appended
-    const right = state.cursorCleared ? '' : destValue.slice(replaceEnd ?? 0).trimRight()
-    const newValue = left + right
-
-    const offset = getTextContentFromHTML(left).length
-
+    const replacedDestValue = state.cursorCleared
+      ? ''
+      : destValue.slice(0, replaceStart || 0) + destValue.slice(replaceEnd || 0)
+    const caretPosition = replaceStart || selection.offset() || 0
```

Comment:
Can you use `state.cursorOffset` instead of `selection.offset()`? I'm not sure if it will work, but it would be nice to not need the `selection` dependency.

Reply:
> It's not quite working for me yet.

Yes I saw this. The reason being selection.offset() is not returning correct caret position when caret is placed at the end of string by hitting backspace. Technically it should have provided the value that is equal to the length of string in focused thought. I am still looking at this and finding how do i correctly track the caret position.